core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMap.java

```diff
@@ -186,23 +193,28 @@ public void init(DataMapModel dataMapModel) throws IOException, MemoryException
         }
       }
     }
-    if (unsafeMemoryDMStore != null) {
-      unsafeMemoryDMStore.finishWriting();
-    }
     if (null != unsafeMemorySummaryDMStore) {
       addTaskSummaryRowToUnsafeMemoryStore(
           summaryRow,
           schemaBinary,
           filePath,
           fileName,
           segmentId);
-      unsafeMemorySummaryDMStore.finishWriting();
     }
     LOGGER.info(
         "Time taken to load blocklet datamap from file : " + dataMapModel.getFilePath() + "is " + (
             System.currentTimeMillis() - startTime));
   }
 
+  @Override public void commit() throws MemoryException, IOException {
```

Comment:
This is specific to BlockletDataMap so lets only keep this class not at interface level.
And also I am not sure why we should separate a commit method. Is there any way to avoid this method?